program/tests/functional.rs

```diff
@@ -480,3 +483,50 @@ async fn close_account_fail_unsigned() {
         TransactionError::InstructionError(0, InstructionError::MissingRequiredSignature)
     );
 }
+
+#[tokio::test]
+async fn validate_owner_success() {
+    let authority = Keypair::new();
+
+    let (solid_pubkey, mut solid_account) = create_solid_account(authority.pubkey()).await;
+    let solid_account_info = (&solid_pubkey, false, &mut solid_account).into_account_info();
+
+    let mut empty_account = Account::new(0, 0, &authority.pubkey());
+    let authority_key = &authority.pubkey();
+    let authority_account_info = (authority_key, true, &mut empty_account).into_account_info();
+
+    let validation_result = validate_owner(&solid_account_info, &[authority_account_info]);
+    assert_eq!(validation_result, Ok(()));
+}
+
+#[tokio::test]
+async fn validate_owner_failed_non_signer() {
+    let authority = Keypair::new();
+
+    let (solid_pubkey, mut solid_account) = create_solid_account(authority.pubkey()).await;
+    let solid_account_info = (&solid_pubkey, false, &mut solid_account).into_account_info();
+
+    let mut empty_account = Account::new(0, 0, &authority.pubkey());
+    let authority_key = &authority.pubkey();
+    let authority_account_info = (authority_key, false, &mut empty_account).into_account_info();
+
+    let validation_result = validate_owner(&solid_account_info, &[authority_account_info]);
+    assert_eq!(validation_result, Err(ProgramError::Custom(0)))
+}
+
+async fn create_solid_account<'a>(authority_pubkey: Pubkey) -> (Pubkey, Account) {
```

Comment:
nit: Since you're not dealing with references, you don't need this lifetime specifier  `<'a>` anymore

Reply:
Fixed